firmware/application/io_file.hpp

```diff
@@ -37,13 +37,15 @@ class FileReader : public stream::Reader {
     FileReader(FileReader&& file) = delete;
     FileReader& operator=(FileReader&&) = delete;
 
-    Optional<File::Error> open(const std::filesystem::path& filename) {
-        return file_.open(filename);
-    }
+    Optional<File::Error> open(const std::filesystem::path& filename);
+    Optional<File::Error> open(const std::filesystem::path& filename, bool do_c8_conversion);
 
     File::Result<File::Size> read(void* const buffer, const File::Size bytes) override;
     const File& file() const& { return file_; }
 
+    void c8_to_c16(const void* buffer, File::Size bytes);
```

Comment:
minor: from a design standpoint it seems like this class shouldn't know about C8/C16 etc. It's more of a generic file type. Consider a wrapper type (like WAVFileReader) for the capture files if you have C8/C16 specific reading logic.

Reply:
I've moved the conversion stuff to new modules io_convert.cpp/hpp, rolling back all changes to this file